src/cider/exec/nextgen/function/CiderStringFunctions.cpp

```diff
@@ -154,3 +156,97 @@ extern "C" RUNTIME_EXPORT int64_t cider_ascii_upper(int8_t* string_heap_ptr,
 extern "C" void test_to_string(int value) {
   std::printf("test_to_string: %s\n", std::to_string(value).c_str());
 }
+
+extern "C" RUNTIME_EXPORT int64_t cider_concat(char* string_heap_ptr,
+                                               const char* lhs,
+                                               int lhs_len,
+                                               const char* rhs,
+                                               int rhs_len) {
+  StringHeap* ptr = reinterpret_cast<StringHeap*>(string_heap_ptr);
```

Comment:
hint: if it's well to inline the function. better to add key words ALWAYS_INLINE.